src/service/translate.js

```diff
@@ -1507,7 +1507,7 @@ function $translate($STORAGE_KEY, $windowProvider, $translateSanitizationProvide
         if (fallbackLanguageIndex < $fallbackLanguage.length) {
           var langKey = $fallbackLanguage[fallbackLanguageIndex];
           result = getFallbackTranslationInstant(langKey, translationId, interpolateParams, Interpolator);
-          if (!result) {
+          if (!result && result !== '') {
```

Comment:
We're dealing with strings so `0`, `undefined`, `null`, `false` and `NaN` are all invalid IMO.